runners/direct-java/src/main/java/org/apache/beam/runners/direct/CopyOnAccessInMemoryStateInternals.java

```diff
@@ -355,6 +379,12 @@ private boolean containedInUnderlying(StateNamespace namespace, StateTag<? super
           public <KeyT, ValueT> MapState<KeyT, ValueT> bindMap(
               StateTag<? super K, MapState<KeyT, ValueT>> address,
               Coder<KeyT> mapKeyCoder, Coder<ValueT> mapValueCoder) {
+            if (mapKeyCoder == null || mapValueCoder == null) {
```

Comment:
Are these needed in all runners though?

My inclination is to say no, which is why I specifically throw exceptions in direct runner since it seems to try to force coder checks wherever encoding can happen.

Runners might handle state in a way which does not need encoding (For example, direct runner works without these coders at all, everything is in-mem. I added these exceptions to make it acknowledge the coders' very existence).